languages/go/src/main/kotlin/io/vrap/codegen/languages/go/client/ClientFileProducer.kt

```diff
@@ -0,0 +1,321 @@
+            relativePath = "$basePackageName/client.go",
+            content = """|
+                |$goGeneratedComment
+                |package $basePackageName
+                |
+                |import (
+                |    "context"
+                |    "io"
+                |    "fmt"
+                |    "net/http"
+                |    "net/url"
+                |    "runtime"
+                |    "golang.org/x/oauth2"
+                |    "golang.org/x/oauth2/clientcredentials"
+                |)
+                |
+                |// Version identifies the current library version. Should match the git tag
+                |const Version = "0.1.0"
+                |
+                |type Client struct {
+                |    httpClient *http.Client
+                |    url        string
+                |    logLevel   int
+                |    userAgent  string
+                |}
+                |
+                |type ClientConfig struct {
+                |    URL         string
+                |    Credentials *clientcredentials.Config
+                |    LogLevel    int
+                |    HTTPClient  *http.Client
+                |    UserAgent   string
+                |}
+                |
+                |// NewClient creates a new client based on the provided ClientConfig
+                |func NewClient(cfg *ClientConfig) (*Client, error) {
+                |
+                |    // If a custom httpClient is passed use that
+                |    var httpClient *http.Client
+                |    if cfg.HTTPClient != nil {
+                |        httpClient = cfg.Credentials.Client(
+                |            context.WithValue(oauth2.NoContext, oauth2.HTTPClient, cfg.HTTPClient))
+                |    } else {
+                |        httpClient = cfg.Credentials.Client(context.TODO())
+                |    }
+                |
+                |    var userAgent = cfg.UserAgent
+                |    if (userAgent == "") {
+                |        userAgent = GetUserAgent()
+                |    }
+                |
+                |    client := &Client{
+                |        url:        cfg.URL,
+                |        logLevel:   cfg.LogLevel,
+                |        httpClient: httpClient,
+                |        userAgent:  userAgent,
+                |    }
+                |
+                |    return client, nil
+                |}
+                |
+                |func (c *Client) get(ctx context.Context, url string, queryParams url.Values, headers http.Header) (*http.Response, error) {
+                |    return c.execute(ctx, "GET", c.url + url, queryParams, headers, nil)
+                |}
+                |
+                |func (c *Client) post(ctx context.Context, url string, queryParams url.Values, headers http.Header, body io.Reader) (*http.Response, error) {
+                |    return c.execute(ctx, "POST", c.url + url, queryParams, headers, body)
```

Comment:
Actually it would make more sense and be more correct to use then `c.url.parse(url)`

reason for this is that it will avoid issues with trailing and leading slash in URLs and the parse method should resolve it correctly whereas a simple `+` will result in invalid URLs